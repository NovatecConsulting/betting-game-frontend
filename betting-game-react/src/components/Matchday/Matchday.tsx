import React from 'react';

const Matchday: React.FC = () => (
  <div
    className='min-h-screen min-w-screen bg-white text-center text-6xl text-black grid items-center tracking-wider'
    data-testid='Matchday'
  >
    Matchday Component
  </div>
);

export default Matchday;
