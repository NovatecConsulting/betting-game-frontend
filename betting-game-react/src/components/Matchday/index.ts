export { default as Matchday } from "./Matchday";
