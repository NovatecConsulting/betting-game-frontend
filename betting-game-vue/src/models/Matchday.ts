import Match from './Match'

export default interface Matchday {
  id: number
  name: String
  firstMatchStartDateTime: string
  lastMatchStartDateTime: string
  matches?: Match[]
}
