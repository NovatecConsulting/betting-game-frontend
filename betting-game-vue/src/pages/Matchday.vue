<template>
  <div class="matchday main-content min-w-screen bg-gray-100 dark:bg-gray-800">
    <div class="matchday__table sm:px-8 md:py-4 overflow-x-auto">
      <MatchdayTable class="m-auto" />
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import MatchdayTable from '../components/MatchdayTable.vue'

@Component({
  components: { MatchdayTable }
})
export default class Matchday extends Vue {}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped></style>
