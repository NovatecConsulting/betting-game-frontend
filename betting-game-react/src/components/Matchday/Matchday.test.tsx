import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Matchday from './Matchday';

describe('Component that displays the match days', () => {
  test('it should mount', () => {
    render(<Matchday />);

    const matchday = screen.getByTestId('Matchday');

    expect(matchday).toBeInTheDocument();
  });
});
