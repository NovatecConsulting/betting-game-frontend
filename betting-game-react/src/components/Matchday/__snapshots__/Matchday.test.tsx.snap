// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matchday component should render and match snapshot with correct content text. 1`] = `
<div
  className="min-h-screen min-w-screen bg-white text-center text-6xl text-black grid items-center tracking-wider"
  data-testid="Matchday"
>
  Matchday Component
</div>
`;
